networkx/algorithms/smallworld.py

```diff
@@ -161,6 +173,8 @@ def lattice_reference(G, niter=5, D=None, connectivity=True, seed=None):
 
     if len(G) < 4:
         raise nx.NetworkXError("Graph has less than four nodes.")
+    if len(G.edges) < 2:
+        raise nx.NetworkXError("Graph has less than two edges")
```

Comment:
Grammatical nit - it wouldn't hurt to update the fewer-than-4-nodes exception message as well. Not a blocker though!

```suggestion
        raise nx.NetworkXError("Graph has fewer than two edges")
```